components/vision.py

```diff
@@ -1,36 +1,93 @@
 import math
+import time
 
+from collections import OrderedDict
+
+from networktables import NetworkTables
 from networktables.util import ntproperty
 
+from pyswervedrive.chassis import SwerveChassis
+
 
 class Vision:
-    target_tape_error = ntproperty("/vision/target_tape_error", math.nan)
-    within_deposit_range = ntproperty("/vision/within_deposit_range", False)
-    ground_tape_x = ntproperty("/vision/ground_tape_x", math.nan)
-    ground_tape_y = ntproperty("/vision/ground_tape_y", math.nan)
-    ground_tape_angle = ntproperty("/vision/ground_tape_angle", math.nan)
+
+    chassis: SwerveChassis
+
+    fiducial_x = ntproperty("/vision/fiducial_x", 0.0, writeDefault=False)
+    fiducial_y = ntproperty("/vision/fiducial_y", 0.0, writeDefault=False)
+    fiducial_time = ntproperty("/vision/fiducial_time", -1.0, writeDefault=False)
+    ping_time = ntproperty("/vision/ping", 0.0, writeDefault=False)
+    raspi_pong_time = ntproperty("/vision/raspi_pong", 0.0, writeDefault=False)
+    rio_pong_time = ntproperty("/vision/rio_pong", 0.0, writeDefault=False)
     # NOTE: x and y are relative to the robot co-ordinate system, not the camera
 
-    def get_target_tape_error(self):
-        if not -1 <= self.target_tape_error <= 1:
-            return None
-        else:
-            return self.target_tape_error
-
-    def get_ground_tape_x(self):
-        if not -1 <= self.ground_tape_x <= 1:
-            return None
-        else:
-            return self.ground_tape_x
-
-    def get_ground_tape_y(self):
-        if not -1 <= self.ground_tape_y <= 1:
-            return None
-        else:
-            return self.ground_tape_y
-
-    def get_ground_tape_angle(self):
-        if not -1 <= self.ground_tape_angle <= 1:
-            return None
-        else:
-            return self.ground_tape_angle
+    def __init__(self):
+        self.latency = 0.0
+        self.last_pong = time.time()
+        self.odometry = OrderedDict()
+
+    def execute(self):
```

Comment:
iirc the vision class isn't currently a component, we would have to make it one if we want to access the execute method

Reply:
Yup, notice that the execute method isn't covered in the tests.